packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeFeeRegistry.sol

```diff
@@ -0,0 +1,240 @@
+        public
+        override defaultChainAirnodePrice;
+
+    /// @notice The default price of an endpoint on an airnode
+    mapping(address => uint256) public override defaultAirnodePrice;
+
+    /// @notice The default price of an endpoint for a chain
+    mapping(uint256 => uint256) public override defaultChainPrice;
+
+    /// @notice The global default price of an endpoint
+    uint256 public override defaultPrice;
+
+    /// @notice A flag that indicates which price to default to in
+    /// the getAirnodeEndpointFee function
+    mapping(address => bool) public override airnodeEndpointFlag;
+
+    /// @notice Prices will have upto 6 decimal places
+    uint8 public constant DECIMALS = 6;
+
+    /// @param _accessControlRegistry AccessControlRegistry contract address
+    /// @param _adminRoleDescription Admin role description
+    /// @param _manager Manager address
+    constructor(
+        address _accessControlRegistry,
+        string memory _adminRoleDescription,
+        address _manager
+    )
+        AirnodeFeeRegistryRolesWithManager(
+            _accessControlRegistry,
+            _adminRoleDescription,
+            _manager
+        )
+    {}
+
+    /// @notice Called by an airnode flag and price setter to
+    /// set the endpoint flag for an airnode
+    /// @param _airnode The address of the airnode
+    /// @param _status The airnode endpoint flag status
+    function setAirnodeEndpointFlag(address _airnode, bool _status)
+        external
+        override
+    {
+        require(
+            hasAirnodeFlagAndPriceSetterRole(msg.sender),
+            "Not airnode flag and price setter"
+        );
+        require(_airnode != address(0), "Address is zero");
+        airnodeEndpointFlag[_airnode] = _status;
+        emit SetAirnodeEndpointFlag(_airnode, _status, msg.sender);
+    }
+
+    /// @notice Called by a global default price setter to set the
+    /// default price in USD
+    /// @param _price The price in USD
+    function setDefaultPrice(uint256 _price) external override {
+        require(
+            hasGlobalDefaultPriceSetterRoleOrIsManager(msg.sender),
+            "Not global default price setter"
+        );
+        require(_price != 0, "Price is zero");
+        defaultPrice = _price * 10**DECIMALS;
+        emit SetDefaultPrice(_price, msg.sender);
+    }
+
+    /// @notice Called by a global default price setter to set the
+    /// default price on a chain in USD
+    /// @param _chainId The id of the chain
+    /// @param _price The price in USD
+    function setDefaultChainPrice(uint256 _chainId, uint256 _price)
+        external
+        override
+    {
+        require(
+            hasGlobalDefaultPriceSetterRoleOrIsManager(msg.sender),
+            "Not global default price setter"
+        );
+        require(_chainId != 0, "ChainId is zero");
+        require(_price != 0, "Price is zero");
+        defaultChainPrice[_chainId] = _price * 10**DECIMALS;
```

Comment:
- `_price` will already have been multiplied by `10**DECIMALS`, don't do it again
- There are no `chainId` and `_price` parameters in the scope so you can use them, no need for the leading underscores.